core/om/sec_test.go

```diff
@@ -175,3 +177,76 @@ func TestKeyActions(t *testing.T) {
 		}
 	}
 }
+
+func TestSecDataLimit(t *testing.T) {
+
+	env := testhelper.Setup(t)
+
+	formatPath := func(path string) string {
+		return env.Root + path
+	}
+
+	cases := map[string]struct {
+		extraArgs []string
+		dataSize  int64
+	}{
+		"addString": {
+			extraArgs: []string{"add", "--key", "foo/bar", "--value", "fooBarValue"},
+			dataSize:  16,
+		},
+		"add10Kb": {
+			extraArgs: []string{"add", "--key", "foo/10kb", "--from", formatPath("/dev/data10kb.bin")},
+			dataSize:  10 * 1000,
+		},
+		"add950Kb": {
+			extraArgs: []string{"add", "--key", "foo/950kb", "--from", formatPath("/dev/data950kb.bin")},
+			dataSize:  950 * 1000,
+		},
+		"add1Mb": {
+			extraArgs: []string{"add", "--key", "foo/1mb", "--from", formatPath("/dev/data1mb.bin")},
+			dataSize:  1001 * 1000,
+		},
+		"add10Mb": {
+			extraArgs: []string{"add", "--key", "foo/10mb", "--from", formatPath("/dev/data10mb.bin")},
+			dataSize:  10001 * 1000,
+		},
+	}
+
+	getCmd := func(name string) []string {
+		args := []string{"test/sec/sec1"}
+		args = append(args, cases[name].extraArgs...)
+		return args
+	}
+
+	env.InstallFile("../../testdata/nodes_info.json", "var/nodes_info.json")
+	env.InstallFile("../../testdata/cluster.conf", "etc/cluster.conf")
```

Comment:
use another cluster.conf file with smaller value for max_key_size for example 1k